### PR TITLE
[chip, dv] update SPI device testplan

### PR DESCRIPTION
**Pull request by weicaiyang**
Updated as it's clear what we need to test in
`chip_sw_spi_device_pass_through_collision`

Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

**Review by a-will — APPROVED** [review]

**Review by tjaychen — APPROVED** [review]

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 1]
```diff
@@ -195,7 +195,20 @@
       name: chip_sw_spi_device_pass_through_collision
       desc: '''Verify the collisions on driving spi_host is handled properly
 
-            TODO, add detail testplan once we have a conclusion on #5134
+            - Enable upload related interrupts and configure spi_device to passthrough mode.
```
Comment (sriyerg) [inline hw/top_earlgrey/data/chip_testplan.hjson:198]:
```suggestion
            - Enable upload related interrupts and configure the spi_device in passthrough mode.
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 2]
```diff
@@ -195,7 +195,20 @@
...
+            - Configure a command slot to enable upload for a flash program/erase command.
+            - Two parallel threads are operating:
```
Comment (sriyerg) [inline hw/top_earlgrey/data/chip_testplan.hjson:200]:
```suggestion
            - Execute two parallel threads:
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 3]
```diff
@@ -195,7 +195,20 @@
...
+              1. SPI host agent.
+                - Send this command via an upstream SPI host agent, then the agent keeps sending
+                read_status to poll the busy bit.
+                - When busy bit is clear, issue a read command to read data from downstream SPI
```
Comment (sriyerg) [inline hw/top_earlgrey/data/chip_testplan.hjson:204]:
```suggestion
                - When the busy bit is low, issue a read command to read the data from the downstream SPI
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 4]
```diff
@@ -195,7 +195,20 @@
...
+                  port. And check data correctness.
+              2. A SW process.
+                - SW receives an upload interrupt and reads the command in the upload fifo for
```
Comment (sriyerg) [inline hw/top_earlgrey/data/chip_testplan.hjson:207]:
```suggestion
                - SW receives an upload interrupt and reads and verifies the command in the upload fifo.
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 5]
```diff
@@ -195,7 +195,20 @@
...
 
-            TODO, add detail testplan once we have a conclusion on #5134
+            - Enable upload related interrupts and configure spi_device to passthrough mode.
+            - Configure a command slot to enable upload for a flash program/erase command.
+            - Two parallel threads are operating:
+              1. SPI host agent.
+                - Send this command via an upstream SPI host agent, then the agent keeps sending
+                read_status to poll the busy bit.
+                - When busy bit is clear, issue a read command to read data from downstream SPI
+                  port. And check data correctness.
```
Comment (sriyerg) [inline hw/top_earlgrey/data/chip_testplan.hjson:205]:
```suggestion
                  port, and check data correctness.
```

**Review by sriyerg — APPROVED** [review]
LGTM, few nit language improvements.

**Review by eunchan — APPROVED** [review]

**Comment by weicaiyang** [comment]
Thanks @sriyerg. Updated.

**Pull request merged**